dashbuilder/dashbuilder-backend/dashbuilder-services/src/main/java/org/dashbuilder/transfer/DataTransferServicesImpl.java

```diff
@@ -0,0 +1,364 @@
+        this.pluginAddedEvent = pluginAddedEvent;
+        this.navTreeChangedEvent = navTreeChangedEvent;
+        this.navTreeStorage = navTreeStorage;
+    }
+
+    @Override
+    public String doExport() throws Exception {
+        String zipLocation = new StringBuilder()
+            .append(System.getProperty("java.io.tmpdir"))
+            .append(File.separator)
+            .append(FILE_PATH)
+            .append(File.separator)
+            .append(EXPORT_FILE_NAME)
+            .toString();
+
+        new File(zipLocation).getParentFile().mkdirs();
+        FileOutputStream fos = new FileOutputStream(zipLocation);
+        ZipOutputStream zos = new ZipOutputStream(fos);
+
+        zipFileSystem(datasetsFS, zos);
+        zipFileSystem(perspectivesFS, zos);
+        zipFileSystem(navigationFS, zos);
+
+        zipFile(createVersionFile(), "VERSION", zos);
+
+        zos.close();
+        fos.close();
+
+        moveZipToFileSystem(zipLocation, systemFS);
+
+        String exportPath = new StringBuilder()
+            .append(SpacesAPI.Scheme.GIT)
+            .append("://")
+            .append(systemFS.getName())
+            .append(File.separator)
+            .append(FILE_PATH)
+            .append(File.separator)
+            .append(EXPORT_FILE_NAME)
+            .toString();
+
+        return exportPath;
+    }
+
+    @Override
+    public List<String> doImport() throws Exception {
+        List<String> imported = new ArrayList<String>();
+        Path root = systemFS.getRootDirectories().iterator().next();
+        String expectedPath = new StringBuilder()
+            .append(File.separator)
+            .append(FILE_PATH)
+            .append(File.separator)
+            .append(IMPORT_FILE_NAME)
+            .toString();
+
+        Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IOException {
+                if (!path.toString().equalsIgnoreCase(expectedPath)) {
+                    return FileVisitResult.CONTINUE;
+                }
+
+                try {
+                    imported.addAll(importFiles(path));
+                    return FileVisitResult.TERMINATE;
+
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    return FileVisitResult.TERMINATE;
+                }
+            }
+        });
+
+        return imported;
+    }
+
+    private List<String> importFiles(Path path) throws Exception {
+        String tempPath = new StringBuilder()
+            .append(System.getProperty("java.io.tmpdir"))
+            .append(File.separator)
+            .append(FILE_PATH)
+            .append(File.separator)
+            .toString();
+
+        List<String> imported = new ArrayList<String>();
+        File destDir = new File(tempPath);
+        ZipInputStream zis = new ZipInputStream(new FileInputStream(path.toFile()));
```

Comment:
Could you please  `try-with-resources` or just `try-catch-finally` block here? If there is an exception thrown later, this won't get closed

Reply:
done